File: scripts/phone_bridge.py
"""
=============================================================================
 GigSync Physical Phone Bridge (Method 1 - Cellular GSM Gateway)
 Real-time bidirectional audio routing between Android Phone and GigSync AI
=============================================================================
"""

import sys
import json
import urllib.request
import os
import time

GIGSYNC_API = os.environ.get("GIGSYNC_API_URL", "http://localhost:8089/api/ai/voice-call")

print("=" * 65)
print("  GIGSYNC CELLULAR PHONE BRIDGE (METHOD 1)")
print("  Android Phone + SIM + 3.5mm Hardware Splitter Loopback")
print("=" * 65)
print(f"Target Server: {GIGSYNC_API}")
print("Status: Ready to process cellular calls.\n")

def process_call(caller_phone, caller_role, speech_text):
    """
    Sends recognized speech from phone audio to GigSync AI Telephony engine.
    """
    payload = json.dumps({
        "callerPhone": caller_phone,
        "callerRole": caller_role,
        "speechText": speech_text
    }).encode("utf-8")

    req = urllib.request.Request(
        GIGSYNC_API,
        data=payload,
        headers={"Content-Type": "application/json"}
    )

    try:
        with urllib.request.urlopen(req) as response:
            res_data = json.loads(response.read().decode("utf-8"))
            print("\n[CALL DETECTED] From:", caller_phone)
            print(f"[CALLER SPOKE]: \"{speech_text}\"")
            print(f"[AI TOOL EXECUTED]: {res_data.get('toolExecuted')}")
            print(f"[DATABASE ACTION]: {json.dumps(res_data.get('toolResult', {}), indent=2)}")
            print(f"[AI SPOKEN RESPONSE]: \"{res_data.get('spokenResponse')}\"\n")
            return res_data
    except Exception as e:
        print(f"[ERROR] Could not reach GigSync backend: {e}")
        return None

if __name__ == "__main__":
    print("Testing pipeline with a sample incoming worker call...")
    time.sleep(1)
    process_call(
        caller_phone="9845011223",
        caller_role="worker",
        speech_text="I am an electrician. I am available tomorrow from 10 AM to 2 PM."
    )

File: scripts/phone_bridge.js
/* ==========================================================================
   GigSync — Physical Phone & Cellular Audio Bridge (Option 3 / Option 2)
   Connects Real Android Phone Call Audio to GigSync AI Voice Gateway
   ==========================================================================

   HARDWARE SETUP (Option 3 - Analog Loopback Cable):
   1. Connect 3.5mm TRRS Splitter to Android Phone (Headphone Jack or USB-C DAC).
   2. Connect Phone Headphone Out -> Laptop Line-In / USB Sound Card Mic (AI listens to caller).
   3. Connect Laptop Headphone Out -> Phone Microphone In (AI speaks back to caller).
   4. Set Android: Settings -> Accessibility / Calls -> Auto-Answer on Headset.
   5. When anyone calls the SIM number, phone auto-answers, AI talks, updates DB!
   ========================================================================== */

const http = require('node:http');

const GIGSYNC_API = process.env.GIGSYNC_API_URL || 'http://localhost:8089/api/ai/voice-call';

console.log('===============================================================');
console.log(' GigSync Physical Phone Bridge (Cellular Gateway)');
console.log('===============================================================');
console.log('Ready to receive audio from connected Android Phone / Line-In.');
console.log('API Target:', GIGSYNC_API);
console.log('Tip: You can also use the Live In-Browser Phone Simulator directly');
console.log('on the GigSync website by clicking the "📞 AI Phone Call" button!');
console.log('===============================================================');

async function processPhoneAudioTurn(callerPhone, callerRole, recognizedSpeech) {
    try {
        const res = await fetch(GIGSYNC_API, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({
                callerPhone,
                callerRole,
                speechText: recognizedSpeech
            })
        });
        const data = await res.json();
        console.log('\n[CALL INCOMING] From:', callerPhone);
        console.log('[CALLER SAID]:', recognizedSpeech);
        console.log('[AI EXECUTED TOOL]:', data.toolExecuted);
        console.log('[AI SPOKEN RESPONSE]:', data.spokenResponse);
        return data;
    } catch (err) {
        console.error('Phone Bridge Error:', err.message);
    }
}

module.exports = { processPhoneAudioTurn };
